opds2_import.py

```diff
@@ -0,0 +1,830 @@
+        http_get=None,
+        metadata_client=None,
+        content_modifier=None,
+        map_from_collection=None,
+        mirrors=None,
+    ):
+        """Initialize a new instance of OPDS2Importer class.
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param collection: Circulation Manager's collection.
+            LicensePools created by this OPDS2Import class will be associated with the given Collection.
+            If this is None, no LicensePools will be created -- only Editions.
+        :type collection: sqlalchemy.orm.session.Session
+
+        :param data_source_name: Name of the source of this OPDS feed.
+            All Editions created by this import will be associated with this DataSource.
+            If there is no DataSource with this name, one will be created.
+            NOTE: If `collection` is provided, its .data_source will take precedence over any value provided here.
+            This is only for use when you are importing OPDS metadata without any particular Collection in mind.
+        :type data_source_name: str
+
+        :param identifier_mapping: Dictionary used for mapping external identifiers into a set of internal ones
+        :type identifier_mapping: Dict
+
+        :param metadata_client: A SimplifiedOPDSLookup object that is used to fill in missing metadata
+        :type metadata_client: SimplifiedOPDSLookup
+
+        :param content_modifier: A function that may modify-in-place representations (such as images and EPUB documents)
+            as they come in from the network.
+        :type content_modifier: Callable
+
+        :param map_from_collection: Identifier mapping
+        :type map_from_collection: Dict
+
+        :param mirrors: A dictionary of different MirrorUploader objects for different purposes
+        :type mirrors: List[MirrorUploader]
+        """
+        super(OPDS2Importer, self).__init__(
+            db,
+            collection,
+            data_source_name,
+            identifier_mapping,
+            http_get,
+            metadata_client,
+            content_modifier,
+            map_from_collection,
+            mirrors,
+        )
+
+        self._logger = logging.getLogger(__name__)
+
+    def _extract_subjects(self, subject):
+        """Extract a list of SubjectData objects from the webpub-manifest-parser's subject.
+
+        :param subject: Parsed subject object
+        :type subject: Union[
+            str,
+            List[Union[str, core_ast.Subject]],
+            core_ast.Subject
+        ]
+
+        :return: List of subjects metadata
+        :rtype: List[SubjectMetadata]
+        """
+        self._logger.debug(
+            u"Started extracting subject metadata from {0}".format(encode(subject))
+        )
+
+        subject_metadata_list = []
+
+        if is_string(subject):
+            subject_metadata = SubjectData(
+                type=None, identifier=None, name=subject, weight=1
```

Comment:
I don't like the idea of creating a subject with no type. Should this be Subject.TAG as well?

Reply:
Updated the code accordingly